plugins/net_plugin/net_plugin.cpp

```diff
@@ -962,7 +979,10 @@ namespace eosio {
       void handle_message( const packed_transaction& msg ) = delete; // packed_transaction_ptr overload used instead
       void handle_message( packed_transaction_ptr trx );
 
-      void process_signed_block( const block_id_type& id, signed_block_ptr msg, block_state_ptr bsp );
+      // returns calculated number of blocks combined latency
+      uint32_t update_latency(const handshake_message& msg);
+
+      void process_signed_block( const block_id_type& id, signed_block_ptr block, block_state_ptr bsp );
```

Comment:
Just curious, `block_state_ptr` contains a `signed_block_ptr`. Will it be different from the `signed_block_ptr` passed as argument?

Reply:
They should never be different if the `block_state_ptr` is non-null. However, they are both passed in because in some cases the `block_state` has been created and in other cases it will be created from the `signed_block`.